### DISCUSSION TITLE
Two new Matcher features: quantifiable token groups and numeric quantifiers

### DISCUSSION THREAD

**Discussion by slbayer** [explosion/spaCy#8411] [New Features & Project Ideas]
I'd like to encourage the addition of two features to the spaCy Matcher rule formalism:

- Quantifiable token sequences. In Matcher, I can add a quantifier to a token, but not a sequence of tokens. I'd like to be able to say something like `{"GROUP": [{"LOWER": ","} {"IS_DIGIT": True}], "OP": "*"}`. 
- Numeric quantifiers. I'd like to be able to specify an upper and lower bound for the quantifier, as in `{"IS_DIGIT": True, "OP": 4}` (four number tokens in a sequence) or `{"IS_DIGIT": True, "OP": [3, 4]}` (three or four number tokens in a sequence).

**Comment by polm** [comment]
> Quantifiable token sequences.

I don't think that's been proposed before

> Numeric quantifiers. 

There's an open issue for this https://github.com/explosion/spaCy/issues/5603 and we'd be happy to have a PR adding support for it. Someone said they were working on this in April but I don't think we've heard anything from them since then.
